Kernel/FileSystem/FATFS/FileSystem.cpp

```diff
@@ -449,14 +450,22 @@ ErrorOr<u32> FATFS::allocate_cluster()
     return Error::from_errno(ENOSPC);
 }
 
-ErrorOr<void> FATFS::notify_cluster_freed()
+ErrorOr<void> FATFS::notify_clusters_freed(u32 first_freed_cluster, u32 freed_cluster_count)
 {
-    if (m_fat_version == FATVersion::FAT32 && m_fs_info.last_known_free_cluster_count != fs_info_data_unknown)
-        TRY(set_free_cluster_count(m_fs_info.last_known_free_cluster_count + 1));
+    if (m_fat_version == FATVersion::FAT32) {
+        u32 free_cluster_count = (m_fs_info.last_known_free_cluster_count == fs_info_data_unknown) ? fs_info_data_unknown : (m_fs_info.last_known_free_cluster_count + freed_cluster_count);
+        u32 first_free_cluster = (first_freed_cluster < m_fs_info.next_free_cluster_hint || m_fs_info.next_free_cluster_hint == fs_info_data_unknown) ? first_freed_cluster : m_fs_info.next_free_cluster_hint;
```

Comment:
I'm a bit confused here.
The [wikipedia page](https://en.wikipedia.org/wiki/Design_of_the_FAT_file_system#FS_Information_Sector) describes this field as "Number of the most recently known to be allocated data cluster."
But we seem to be treating it as the lowest free cluster number here?
Are clusters always allocated sequentially?